packages/@azure/servicebus/data-plane/lib/queueClient.ts

```diff
@@ -147,20 +147,93 @@ export class QueueClient implements Client {
   }
 
   /**
-   * Creates a Receiver to be used for receiving messages in batches or by registering handlers.
+   * Creates a Receiver for receiving messages from a Queue which does not have sessions enabled.
    * Throws error if an open receiver already exists for this QueueClient.
-   * @param options Options for creating the receiver.
+   *
+   * Throws error if the Queue has sessions enabled.
+   *
+   * @param recieveMode An enum indicating the mode in which messages should be received. Possible
+   * values are `ReceiveMode.peekLock` and `ReceiveMode.receiveAndDelete`
+   *
+   * @returns Promise<Reciever> A promise that resolves to a receiver to receive messages from a
+   * Queue which does not have sessions enabled.
+   */
+  public async createReceiver(recieveMode: ReceiveMode): Promise<Receiver>;
+  /**
+   * Creates a Receiver for receiving messages from a session enabled Queue. When no sessionId is
+   * given, a random session among the available sessions is used.
+   *
+   * Throws error if an open receiver already exists for given sessionId.
+   * Throws error if the Queue does not have sessions enabled.
+   *
+   * @param recieveMode An enum indicating the mode in which messages should be received. Possible
+   * values are `ReceiveMode.peekLock` and `ReceiveMode.receiveAndDelete`
+   * @param sessionOptions Options to provide sessionId and duration of automatic lock renewal for
+   * the session receiver.
+   *
+   * @returns Promise<SessionReciever> A promise that resolves to a receiver to receive from a
+   * session in the Queue.
    */
-  createReceiver(options?: MessageReceiverOptions): Receiver {
+  public async createReceiver(
+    receiveMode: ReceiveMode,
+    sessionOptions: SessionReceiverOptions
+  ): Promise<SessionReceiver>;
+  /**
+   * Create a Receiver for receiving messages from a Queue.
+   *
+   * @param recieveMode An enum indicating the mode in which messages should be received. Possible
+   * values are `ReceiveMode.peekLock` and `ReceiveMode.receiveAndDelete`
+   * @param sessionOptions Applicable only for Queues that have sessions enabled. Use these options
+   * to provide sessionId and duration for which automatic lock renewal for should be done for the
+   * receiver.
+   *
+   * @returns Promise<Receiver|SessionReciever> A promise that resolves to a receiver to receive
+   * from a session in the Queue if `sessionOptions` were provided. Else, the promise resolves to a
+   * receiver to receive messages from the Queue.
+   */
+  public async createReceiver(
```

Comment:
Why do we have 3 methods for `createReceiver`? Is this the common method that the other 2 are calling into?

Reply:
Yes. The first 2 don't have a body. The third is the one that has the entire implementation